js/tools/tooloption.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * Class for generating options for tools. Types of options are specified as string arguments in the constructor
+ * and mapped to functions that return DOM elements that are displayed in the tool customization panel on the right.
+ * This object is intended to be applied in an "options" key on the class representing that tool.
+ * 
+ * For example:
+ * rectangle.js contains the Rectangle class. At the very bottom, Rectangle.options is set to a new ToolOption
+ * with "fill" and "stroke" passed into its constructor. Any associated values in the Rectangle class that needs
+ * to be altered with those options can be set to values in the Rectangle.options object.
+ */
+
+class ToolOption {
+    constructor(...options) {
+        this.options = options;
+        this.optionMapping = {
+            "fill": () => this.fillOptions(),
+            "stroke": () => this.strokeOptions()
+        }
+
+        this.optionDOMElements = [];
+
+        this.options.forEach(enabledOption => {
+            this.optionDOMElements.push(this.optionMapping[enabledOption]());
+        });
+    }
+
+    fillOptions() {
```

Comment:
Is there an advantage to reconstructing the options html every time? Does binding not work?